### ISSUE TITLE
--fontsource-display variable does not work

### ISSUE THREAD

**Issue by schmist** [DecliningLotus/fontsource#726]
### Describe the bug

CSS variables (as `--fontsource-display` described [here](https://fontsource.org/docs/getting-started/display)) seem not to work when used in an `@font-face` statement. See also [here](https://stackoverflow.com/questions/58752373/custom-css-properties-in-font-face.).

### Steps to Reproduce

As described here: https://fontsource.org/docs/getting-started/display. Declared variable is not honored.

### Expected behavior

Declared variable should be honored (seems not possible).

### Version

5.0.1

### OS

Linux

### Browser

Chrome

**Comment by ayuhito** [comment]
This is extremely unfortunate since it seems like no browsers seem to support custom properties in `@font-face` for undocumented reasons. Yet they support it in selectors like `@keyframes` 🤷 

I think we'll have to revert this change to use `swap` as a default. I still do want to offer an escape hatch, so once I have time I could investigate if there is an option where users can write their own `@font-face` statements and directly link to the font files themselves.

e.g. does referencing `node_modules` like this work in Webpack or Vite? Will it successfully rewrite the URLs?
```css
@font-face {
  font-family: 'Abel';
  font-style: normal;
  font-display: optional;
  font-weight: 400;
  src: url(/node_modules/@fontsource/abel/files/abel-latin-400-normal.woff2) format('woff2'), url(/node_modules/@fontsource/abel/files/abel-latin-400-normal.woff) format('woff');
}
```

The URL might also be prefixed with `@` or `~`, I'm not too sure.

**Comment by schmist** [comment]
To be able to set the `font-display` property, a user could use the following mixin and declare the font manually without the index.css import:

```
@mixin font-face($font-name, $font-id) {
  $weight: 400;
  $style: normal;
  $font-dir: "~@fontsource/#{$font-id}/files";
  $woff-path: "#{$font-dir}/#{$font-id}-latin-#{$weight}-#{$style}.woff";
  $woff2-path: "#{$font-dir}/#{$font-id}-latin-#{$weight}-#{$style}.woff2";

  @font-face {
    font-family: "#{$font-name}";
    font-style: $style;
    font-display: block;
    font-weight: $weight;
    src: url("#{$woff-path}") format("woff2"), url("#{$woff2-path}") format("woff");
  }
}
```

**Comment by stormynight9** [comment]
I'm using Astro + TailwindCSS, how do I add `font-display: swap`?
Lighthouse is giving this message: _Ensure text remains visible during webfont load_ which I believe what causing it is the font-display property not applying swap as value.

**Comment by maxstoyanov** [comment]
Are there any plans to revert back to swap in the meantime?

**Comment by ayuhito** [comment]
> Are there any plans to revert back to swap in the meantime?

It should be already reverted in #751. Is this still present?

**Comment by maxstoyanov** [comment]
> > Are there any plans to revert back to swap in the meantime?
>
> It should be already reverted in #751. Is this still present?

Never mind, you are correct. It appears that something else was messing with my font loading.
I just checked the npm build and everything is correct. The issue can be closed then.

**Comment by ayuhito** [comment]
Closed in #954. This is just an escape hatch that simplifies writing custom `@font-face` declarations by auto-generating the CSS for you.
